File: 3601-find-the-k-th-character-in-string-game-ii/3601-find-the-k-th-character-in-string-game-ii.py
class Solution:
    def kthCharacter (self, k : int, operations : List [int]) -> str:
        c = 0
        k -= 1
        i = 0
        while k != 0:
            c += int ((k & 1) & operations [i])
            i += 1
            k >>= 1
        return chr ((c % 26) + ord ("a"))

File: 3601-find-the-k-th-character-in-string-game-ii/3601-find-the-k-th-character-in-string-game-ii.c
char kthCharacter (long long k, int* operations, int operationsSize) {
    int c = 0; k--;
    for (int i = 0; k != 0; i++, k >>= 1) c += ((int) (k & 1) & operations [i]);
    return (char) ((c % 26) + 'a');
}